airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/DogStatsDMetricClient.java

```diff
@@ -83,34 +85,44 @@ public void count(final MetricsRegistry metric, final long amt, final String...
    *
    * @param metric
    * @param val to record.
-   * @param tags
+   * @param attributes
    */
   @Override
-  public void gauge(final MetricsRegistry metric, final double val, final String... tags) {
+  public void gauge(final MetricsRegistry metric, final double val, final MetricAttribute... attributes) {
     if (instancePublish) {
       if (statsDClient == null) {
         // do not loudly fail to prevent application disruption
         log.warn("singleton not initialized, gauge {} not emitted", metric);
         return;
       }
 
-      log.info("publishing gauge, name: {}, value: {}, tags: {}", metric, val, tags);
-      statsDClient.gauge(metric.getMetricName(), val, tags);
+      log.info("publishing gauge, name: {}, value: {}, attributes: {}", metric, val, attributes);
+      statsDClient.gauge(metric.getMetricName(), val, toTags(attributes));
     }
   }
 
   @Override
-  public void distribution(MetricsRegistry metric, double val, final String... tags) {
+  public void distribution(final MetricsRegistry metric, final double val, final MetricAttribute... attributes) {
     if (instancePublish) {
       if (statsDClient == null) {
         // do not loudly fail to prevent application disruption
         log.warn("singleton not initialized, distribution {} not emitted", metric);
         return;
       }
 
-      log.info("recording distribution, name: {}, value: {}, tags: {}", metric, val, tags);
-      statsDClient.distribution(metric.getMetricName(), val, tags);
+      log.info("recording distribution, name: {}, value: {}, attributes: {}", metric, val, attributes);
+      statsDClient.distribution(metric.getMetricName(), val, toTags(attributes));
     }
   }
 
+  /**
+   * Converts each {@link MetricAttribute} tuple to a list of tags consumable by StatsD.
+   *
+   * @param attributes An array of {@link MetricAttribute} tuples.
+   * @return An array of tag values.
+   */
+  private String[] toTags(final MetricAttribute... attributes) {
+    return Stream.of(attributes).map(a -> a.value()).collect(Collectors.toList()).toArray(new String[] {});
```

Comment:
Shouldn't we try to keep the `a.key()` as well?

Reply:
I agree, it is confusing to remove it. It feels that it is useless since it is never send to data dog.